### PR TITLE
cat_wallet mypy fixes

### PR DESCRIPTION
**Pull request by arvidn**
This patch removes the mypy exceptions for the cat wallet files (but not the tests) and fixes type annotations

### CODE REVIEW

**Review by matt-o-how — DISMISSED** [review]

**Review by mariano54 — DISMISSED** [review]

File: chia/wallet/cat_wallet/cat_utils.py [hunk 1]
```diff
@@ -36,15 +36,16 @@ def match_cat_puzzle(mod: Program, curried_args: Program) -> Optional[Iterator[P
     if it is, return the curried arguments
     """
     if mod == CAT_MOD:
-        return curried_args.as_iter()
+        ret: Iterator[Program] = curried_args.as_iter()
```
Comment (altendky) [inline chia/wallet/cat_wallet/cat_utils.py:39]: Note that `Program` is `Any` due to inheriting from `SExp` which is in the unhinted `clvm` and thus treated as `Any`.  https://github.com/Chia-Network/clvm/pull/102  No need to change here, just be aware that the hinting doesn't do much.

File: chia/wallet/cat_wallet/cat_utils.py [hunk 2]
```diff
@@ -82,7 +83,8 @@ def subtotals_for_deltas(deltas) -> List[int]:
 def next_info_for_spendable_cat(spendable_cat: SpendableCAT) -> Program:
     c = spendable_cat.coin
     list = [c.parent_coin_info, spendable_cat.inner_puzzle.get_tree_hash(), c.amount]
-    return Program.to(list)
+    ret: Program = Program.to(list)
+    return ret
```
Comment (altendky) [inline chia/wallet/cat_wallet/cat_utils.py:87]: I don't like re-hinting return values in general, but especially not when they hide the underlying issue.
```suggestion
    # ignoring hint error here for:
    # https://github.com/Chia-Network/clvm/pull/102
    return Program.to(list)  # type: ignore[no-any-return]
```

File: chia/wallet/cat_wallet/cat_wallet.py [hunk 3]
```diff
@@ -54,11 +54,14 @@
...
 
+if TYPE_CHECKING:
+    from chia.wallet.wallet_state_manager import WalletStateManager
+
 # This should probably not live in this file but it's for experimental right now
 
 
 class CATWallet:
-    wallet_state_manager: Any
+    wallet_state_manager: "WalletStateManager"
```
Comment (altendky) [inline chia/wallet/cat_wallet/cat_wallet.py:64]: Also add `from __future__ import annotations` to the top of the file.  https://gist.github.com/altendky/7c423a63d5091b2ccf2c418624317f93#forward-references

```suggestion
    wallet_state_manager: WalletStateManager
```

File: chia/wallet/cat_wallet/cat_wallet.py [hunk 4]
```diff
@@ -72,8 +75,12 @@ def default_wallet_name_for_unknown_cat(limitations_program_hash_hex: str) -> st
 
     @staticmethod
     async def create_new_cat_wallet(
-        wallet_state_manager: Any, wallet: Wallet, cat_tail_info: Dict[str, Any], amount: uint64, name=None
-    ):
+        wallet_state_manager: "WalletStateManager",
```
Comment (altendky) [inline chia/wallet/cat_wallet/cat_wallet.py:78]:
```suggestion
        wallet_state_manager: WalletStateManager,
```

File: chia/wallet/cat_wallet/cat_wallet.py [hunk 5]
```diff
@@ -162,10 +169,10 @@ async def create_new_cat_wallet(
 
     @staticmethod
     async def create_wallet_for_cat(
-        wallet_state_manager: Any,
+        wallet_state_manager: "WalletStateManager",
```
Comment (altendky) [inline chia/wallet/cat_wallet/cat_wallet.py:172]:
```suggestion
        wallet_state_manager: WalletStateManager,
```

File: chia/wallet/cat_wallet/cat_wallet.py [hunk 6]
```diff
@@ -199,10 +206,10 @@ async def create_wallet_for_cat(
     @classmethod
     async def create_from_puzzle_info(
         cls,
-        wallet_state_manager: Any,
+        wallet_state_manager: "WalletStateManager",
```
Comment (altendky) [inline chia/wallet/cat_wallet/cat_wallet.py:209]:
```suggestion
        wallet_state_manager: WalletStateManager,
```

File: chia/wallet/cat_wallet/cat_wallet.py [hunk 7]
```diff
@@ -213,7 +220,7 @@ async def create_from_puzzle_info(
 
     @staticmethod
     async def create(
-        wallet_state_manager: Any,
+        wallet_state_manager: "WalletStateManager",
```
Comment (altendky) [inline chia/wallet/cat_wallet/cat_wallet.py:223]:
```suggestion
        wallet_state_manager: WalletStateManager,
```

File: chia/wallet/cat_wallet/lineage_store.py [hunk 8]
```diff
@@ -19,7 +19,7 @@ class CATLineageStore:
     table_name: str
 
     @classmethod
-    async def create(cls, db_wrapper: DBWrapper2, asset_id: str):
+    async def create(cls, db_wrapper: DBWrapper2, asset_id: str) -> "CATLineageStore":
```
Comment (altendky) [inline chia/wallet/cat_wallet/lineage_store.py:22]:
```suggestion
    async def create(cls, db_wrapper: DBWrapper2, asset_id: str) -> CATLineageStore:
```

**Review by altendky — CHANGES_REQUESTED** [review]

File: chia/wallet/cat_wallet/lineage_store.py (hunk 8 above)
Comment (arvidn) [reply chia/wallet/cat_wallet/lineage_store.py:22]: this change breaks flake8. `CATLineageStore` is not defined yet at this point

Comment (altendky) [reply chia/wallet/cat_wallet/lineage_store.py:22]: `from __future__ import annotations`

Comment (arvidn) [reply chia/wallet/cat_wallet/lineage_store.py:22]: I see. I looked in the wrong file, I thought we had this already

**Review by altendky — APPROVED** [review]

**Review by mariano54 — APPROVED** [review]

**Pull request merged**